### PR TITLE
 Closes #9972: Invoke Push feature's registerDeviceObserver call on the main thread

### LINKED ISSUE #9972: FxaDeviceConstellation.registerDeviceObserver: java.lang.IllegalStateException: Method addObserver must be called on the main thread
Using a newer Android X version, I see this startup crash. Looks like the `enforceMainThreadIfNeeded()` is new.

```
         AndroidRuntime  E  FATAL EXCEPTION: main
                         E  Process: org.mozilla.reference.browser.debug, PID: 16312
                         E  java.lang.IllegalStateException: Method addObserver must be called on the main thread
                         E      at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(LifecycleRegistry.java:317)
                         E      at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:172)
                         E      at mozilla.components.support.base.observer.ObserverRegistry.register(ObserverRegistry.kt:66)
                         E      at mozilla.components.service.fxa.FxaDeviceConstellation.registerDeviceObserver(FxaDeviceConstellation.kt:126)
                         E      at mozilla.components.feature.accounts.push.AccountObserver.onAuthenticated(FxaPushSupportFeature.kt:160)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager$accountStateSideEffects$6.invoke(FxaAccountManager.kt:535)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager$accountStateSideEffects$6.invoke(FxaAccountManager.kt:123)
                         E      at mozilla.components.support.base.observer.ObserverRegistry.notifyObservers(ObserverRegistry.kt:136)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager.notifyObservers(Unknown Source:7)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager.accountStateSideEffects(FxaAccountManager.kt:535)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager.stateActions(FxaAccountManager.kt:755)
                         E      at mozilla.components.service.fxa.manager.FxaAccountManager$processQueue$2.invokeSuspend(FxaAccountManager.kt:506)
                         E      at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                         E      at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                         E      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                         E      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                         E      at java.lang.Thread.run(Thread.java:923)
```

### PR DESCRIPTION
**Pull request by grigoryk**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by grigoryk** [comment]
I've tested this against https://github.com/pocmo/reference-browser/tree/compose - built against this PR, it no longer crashes on startup when logged-in to FxA.

File: components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt [hunk 1]
```diff
@@ -157,7 +157,9 @@ internal class AccountObserver(
         // registration could happen after onDevicesUpdate has been called, without having to tie this
         // into the account "auth lifecycle".
         // See https://github.com/mozilla-mobile/android-components/issues/8766
-        account.deviceConstellation().registerDeviceObserver(constellationObserver, lifecycleOwner, autoPause)
+        CoroutineScope(Dispatchers.Main).launch {
```
Comment (pocmo) [inline components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt:160]: nit: If you do not keep a reference to the scope, to limit its lifetime, then you could use `GlobalScope`:

```GlobalScope.launch(Dispatchers.Main) {  }```

**Review by pocmo — APPROVED** [review]
Thank you for fixing this for me :)

File: components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt (hunk 1 above)
Comment (grigoryk) [reply components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt]: What I do like about this is that all instances of this pattern are then clearly identified by usages of `GlobalScope`. Makes it easy to find and reassess if we need to manage these scopes better.

**Pull request merged**